testing/integration/upgrade_test.go

```diff
@@ -327,6 +340,37 @@ func (s *StandaloneUpgradeTestSuite) TestUpgradeStandaloneElasticAgentToSnapshot
 	// s.Require().Equal(expectedAgentHashAfterUpgrade, version.Commit, "agent commit hash changed after upgrade")
 }
 
+func extractCommitHashFromArtifact(t *testing.T, ctx context.Context, artifactVersion *version.ParsedSemVer, agentProject tools.Project) string {
+	tmpDownloadDir := t.TempDir()
+
+	operatingSystem := runtime.GOOS
+	architecture := runtime.GOARCH
+	suffix, err := atesting.GetPackageSuffix(operatingSystem, architecture)
+	require.NoErrorf(t, err, "error determining suffix for OS %q and arch %q", operatingSystem, architecture)
+	prefix := fmt.Sprintf("elastic-agent-%s", artifactVersion.VersionWithPrerelease())
+	pkgName := fmt.Sprintf("%s-%s", prefix, suffix)
+	require.Containsf(t, agentProject.Packages, pkgName, "Artifact API response does not contain pkg %s", pkgName)
+	artifactFilePath := filepath.Join(tmpDownloadDir, pkgName)
```

Comment:
The work being done here is mostly not specific to this test.  WDYT about leaving `atesting.downloadPackage` as-is and creating a new `atesting.DownloadPackage` function that takes the package version and download path as arguments, does all the logic above, and calls `downloadPackage` internally?

I could see this new `atesting.DownloadPackage` being useful to other tests too.

Reply:
The choice of making the DownloadPackage function public comes from the fact  that it already does reusable work...
As for the logic above, it relies on the artifact API which I am not sure that we want in all the cases so for the moment I would keep it as is, we can always refactor in a follow-up PR

Reply:
> As for the logic above, it relies on the artifact API which I am not sure that we want in all the cases so for the moment I would keep it as is, we can always refactor in a follow-up PR

Maybe I'm not seeing it, but where does the logic above (lines 346-353) rely on the artifact API?

Reply:
Ah, never mind. I see the issue now.  Good to keep this as-is.